core/src/Streamly/Internal/FileSystem/Posix/ReadDir.hsc

```diff
@@ -360,7 +529,8 @@ readEitherChunks alldirs =
                             (ChunkStreamLoop curdir xs dirp [] 0 files nfiles)
                          else return $ Skip
                             (ChunkStreamLoop curdir xs dirp dirs1 ndirs1 files nfiles)
-            else do
+                GSIsMetaDir ->  return $ Skip st
```

Comment:
We can use a wildcard case at the end for skipping. In fact we can even combine the two cases of GSIsMetaDir and GSIsNotDir in to a single GSIgnoreEnt.